### PR TITLE
feat: support analysing AMD module format

### LINKED ISSUE #4313: support analyze AMD module format
### What problem does this feature solve?

- #2787

### What does the proposed API of configuration look like?

https://webpack.js.org/configuration/other-options/#amd

### PR DESCRIPTION
**Pull request by nilptr**
## Summary

resolve #4313, support analysing AMD module format (context is not supported yet)

## Checklist

- [x] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by nilptr** [comment]
hi @GiveMe-A-Name @ahabhgk this PR is ready to review. Please help take a look. Thanks.

File: crates/rspack_core/src/dependency/dependency_type.rs [hunk 1]
```diff
@@ -10,10 +10,10 @@ pub enum DependencyType {
   Unknown,
   ExportInfoApi,
   Entry,
-  // ESM import
+  // Harmony import
```
Comment (ahabhgk) [inline crates/rspack_core/src/dependency/dependency_type.rs:13]:
```suggestion
  // ESM import
```

keep this as is

File: crates/rspack_core/src/dependency/dependency_type.rs [hunk 2]
```diff
@@ -10,10 +10,10 @@ pub enum DependencyType {
...
   EsmImport,
   EsmImportSpecifier,
-  // ESM export
+  // Harmony export
```
Comment (ahabhgk) [inline crates/rspack_core/src/dependency/dependency_type.rs:16]: same here

File: crates/rspack_core/src/dependency/dependency_type.rs (hunk 1 above)
Comment (nilptr) [reply crates/rspack_core/src/dependency/dependency_type.rs]: updated

File: crates/rspack_core/src/dependency/dependency_type.rs (hunk 2 above)
Comment (nilptr) [reply crates/rspack_core/src/dependency/dependency_type.rs]: updated

**Comment by ahabhgk** [comment]
CI should pass

**Comment by nilptr** [comment]
> CI should pass

fixed

**Review by ahabhgk — APPROVED** [review]
Great work! Thanks!

**Comment by nanianlisao** [comment]
That is great!

It might be wise to consider ensuring consistency with Webpack as well.

https://github.com/web-infra-dev/rspack/blob/46d05668afbfc4e2a40d92d0efb2ad07301b2697/packages/rspack/src/config/normalization.ts#L373-L376

https://github.com/webpack/webpack/blob/3919c844eca394d73ca930e4fc5506fb86e2b094/lib/WebpackOptionsApply.js#L389-L395

By the way, the content regarding this configuration has not yet been updated on the official website.

@chenjiahan

**Pull request merged**